analyzers/src/SonarAnalyzer.CSharp/Rules/ExceptionsNeedStandardConstructors.cs

```diff
@@ -68,18 +69,12 @@ private static bool HasStandardConstructors(INamedTypeSymbol classSymbol)
                        KnownType.System_Runtime_Serialization_StreamingContext);
         }
 
-        private static bool HasConstructor(ImmutableArray<IMethodSymbol> constructors,
-            Accessibility accessibility, params KnownType[] expectedParameterTypes)
-        {
-            return constructors.Any(c => IsMatchingConstructor(c, accessibility, expectedParameterTypes));
-        }
+        private static bool HasConstructor(ImmutableArray<IMethodSymbol> constructors, Accessibility accessibility, params KnownType[] expectedParameterTypes) =>
+            constructors.Any(c => IsMatchingConstructor(c, accessibility, expectedParameterTypes));
 
-        private static bool IsMatchingConstructor(IMethodSymbol constructor, Accessibility accessibility,
-            KnownType[] expectedParameterTypes)
-        {
-            return constructor.DeclaredAccessibility == accessibility &&
-                   EnumerableExtensions.Equals(constructor.Parameters, expectedParameterTypes,
-                       (p1, p2) => p1.Type.Is(p2));
-        }
+        private static bool IsMatchingConstructor(IMethodSymbol constructor, Accessibility accessibility, KnownType[] expectedParameterTypes) =>
+            constructor.DeclaredAccessibility == accessibility &&
```

Comment:
The && operator is in the wrong place.

Reply:
yes, I've considered that I should stop the cleanup at some point :)